sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2349,7 +2350,26 @@ class Analyzer(
   }
 
   /**
-   * Replace [[TimeZoneAwareExpression]] without timezone id by its copy with session local
+   * Replace unresolved expressions in grouping keys with resolved ones in SELECT clauses.
+   */
+  object ResolveAggAliasInGroupBy extends Rule[LogicalPlan] {
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case agg @ Aggregate(groups, aggs, child)
+          if conf.groupByAliasesEnabled && child.resolved && groups.exists(!_.resolved) =>
```

Comment:
ISTM all the expressions in SELECT clauses are expected be resolved in Resolution batch. So, I removed the condition from the latest version of this pr. Actually, I couldn't find error test cases without the condition. Any idea for that?